packages/task/src/node/task-configurations.ts

```diff
@@ -0,0 +1,244 @@
+            // task file changed? re-read it
+            if (params.changes.some(c => FileUri.fsPath(c.uri) === this.homedirConfig)) {
+                this.userTasksMap.clear();
+                try {
+                    this.readTasks(this.homedirConfig).then(tasks => {
+                        for (const task of tasks) {
+                            this.userTasksMap.set(task.label, task);
+                        }
+                        resolve();
+                    });
+                } catch (err) {
+                    this.logger.error(err);
+                    reject(err);
+                }
+            }
+
+            if (params.changes.some(c => FileUri.fsPath(c.uri) === this.workspaceConfig)) {
+                this.workspaceTasksMap.clear();
+                try {
+                    this.readTasks(this.workspaceConfig).then(tasks => {
+                        for (const task of tasks) {
+                            this.workspaceTasksMap.set(task.label, task);
+                        }
+                        resolve();
+                    });
+                } catch (err) {
+                    this.logger.error(err);
+                    reject(err);
+                }
+            }
+        });
+    }
+
+    protected readTasks(uri: string): Promise<ITaskOptions[]> {
+        return new Promise((resolve, reject) => {
+            this.fileSystem.exists(uri).then(exists => {
+                if (!exists) {
+                    reject(`tasks.json file doesn't exist: ${uri}`);
+                }
+                this.fileSystem.resolveContent(uri).then(({ stat, content }) => {
+                    const strippedContent = jsoncparser.stripComments(content);
+                    const errors: ParseError[] = [];
+                    const tasks = jsoncparser.parse(strippedContent, errors);
+
+                    if (errors.length) {
+                        for (const error of errors) {
+                            this.logger.error("JSON parsing error", error);
+                        }
+                    }
+                    resolve(this.filterDuplicates(tasks['tasks']));
+                });
+            }).catch(reason => {
+                if (reason) {
+                    this.logger.error(`Failed to read tasks ${uri}:`, reason);
+                    reject(`Failed to read tasks ${uri}: ${reason}`);
+                }
+                reject('Failed to read tasks file');
+            });
+        });
+    }
+
+    private filterDuplicates(tasks: ITaskOptions[]): ITaskOptions[] {
+        const filteredTasks: ITaskOptions[] = [];
+        for (const task of tasks) {
+            if (filteredTasks.some(t => t.label === task.label)) {
+                this.logger.error(`Error parsing tasks.json: found duplicate entry for label ${task.label}`);
+            } else {
+                filteredTasks.push(task);
+            }
+        }
+        return filteredTasks;
+    }
+}
```

Comment:
I think you're right. The difference between the two configs (home, WS) was only in my head: since changing the current workspace (in the client, file->open(), chose new WS) if fact restarts the client, I do not need to actually handle the changing config file for the workspace here, I just need to cleanup when a client goes away and start clean when the next one comes. Done.